### PR TITLE
Story#586

### PR DESCRIPTION
**Pull request by srikanthgudipudi**
Story 586

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -9,12 +9,12 @@ import { HomeComponent } from './home/home.component';
 const routes: Routes = [
   {
     path: 'users',
-    canActivate: [RouteGuard],
+    // canActivate: [RouteGuard],
```
Comment (kiranharidas187) [inline src/app/app-routing.module.ts]: Why is this commented?

File: src/app/home/home.component.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export class HomeComponent implements OnInit {
     if (this.promiseRowData['result']) {
       this.rolesArray = this.promiseRowData['result'].roles;
     }
+    console.log(' this.rolesArray', this.rolesArray);
```
Comment (kiranharidas187) [inline src/app/home/home.component.ts]: Please remove this unwanted changes

File: src/app/home/home.component.ts [hunk 3]
```diff
@@ -85,7 +86,8 @@ export class HomeComponent implements OnInit {
         duration: 10000,
         verticalPosition: 'top'
       });
-    } else {
+    }
```
Comment (kiranharidas187) [inline src/app/home/home.component.ts]: Please remove this unwanted changes

File: src/app/modules/admin-core/guards/route.guard.ts [hunk 4]
```diff
@@ -13,21 +14,22 @@ export class RouteGuard implements CanActivate {
   rolesArray: any;
   tokendetails: any;
   constructor(private usersService: UsersService, private route: Router,
-    private keycloakService: keyCloakService) {
+    private keycloakService: keyCloakService, private commonServiceService: CommonServiceService) {
+
   }
```
Comment (kiranharidas187) [inline src/app/modules/admin-core/guards/route.guard.ts:19]: Can we make use of this same router guard for all the Pages

File: src/app/modules/admin-core/services/organisation-service/organisation.service.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+})
+export class OrganisationService {
+
+  constructor(private Http: HttpClient) { }
+
+  // getting Organisation list 
+  organisationList(data, search) {
+    console.log('organisationList', data);
+    // admin-service/api/v1/organisations/detailList?limit=10&offset=1
+    return this.Http.get(environment.base_url + OrganisationConfig.organisationList + '?limit=' + data.size + '&offset=' + data.page + '&search='+ search);
```
Comment (kiranharidas187) [inline src/app/modules/admin-core/services/organisation-service/organisation.service.ts]: Is this same as the user's list page? I think we were not using offset there

File: src/app/modules/admin-core/services/organisation-service/organisation.service.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  providedIn: 'root'
+})
+export class OrganisationService {
+
+  constructor(private Http: HttpClient) { }
+
+  // getting Organisation list 
+  organisationList(data, search) {
+    console.log('organisationList', data);
+    // admin-service/api/v1/organisations/detailList?limit=10&offset=1
```
Comment (kiranharidas187) [inline src/app/modules/admin-core/services/organisation-service/organisation.service.ts]: Please remove

File: src/app/modules/organisations/organisations-list/organisations-list.component.html [hunk 7]
```diff
@@ -52,19 +52,22 @@
...
-    <mat-table [dataSource]="dataSource" matSort>
 
-      <ng-container matColumnDef="select">
+
+
+    <mat-table *ngIf="dataSource" [dataSource]="dataSource" matSort>
+
+      <!-- <ng-container matColumnDef="select">
         <th mat-header-cell *matHeaderCellDef style="padding-top: 10px;">
           <mat-checkbox (change)="$event ? masterToggle() : null" [checked]="selection.hasValue() && isAllSelected()"
```
Comment (kiranharidas187) [inline src/app/modules/organisations/organisations-list/organisations-list.component.html:72]: Remove all commented code if not required

File: src/app/modules/admin-core/guards/route.guard.ts (hunk 4 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-core/guards/route.guard.ts:19]: only we can differentiate the platform_admin and organisation_admin

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (srikanthgudipudi) [reply src/app/app-routing.module.ts]: With out login we cannot go inside right ?then  differentiating with platform_admin and organisation_admin

File: src/app/home/home.component.ts (hunk 2 above)
Comment (srikanthgudipudi) [reply src/app/home/home.component.ts]: Removed

File: src/app/home/home.component.ts (hunk 3 above)
Comment (srikanthgudipudi) [reply src/app/home/home.component.ts]: Removed

File: src/app/modules/admin-core/services/organisation-service/organisation.service.ts (hunk 6 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-core/services/organisation-service/organisation.service.ts]: Removed

File: src/app/modules/admin-core/services/organisation-service/organisation.service.ts (hunk 5 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-core/services/organisation-service/organisation.service.ts]: No, from the backend api they changed, shall we ask for the same?

File: src/app/modules/organisations/organisations-list/organisations-list.component.html (hunk 7 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/organisations-list/organisations-list.component.html:72]: We are not sure what is required, that is why it is commented ,after discussion i will remove completely

File: src/app/app-routing.module.ts [hunk 8]
```diff
@@ -9,12 +9,11 @@ import { HomeComponent } from './home/home.component';
 const routes: Routes = [
   {
     path: 'users',
-    canActivate: [RouteGuard],
     loadChildren: './modules/users/users.module#UsersModule'
```
Comment (kiranharidas187) [inline src/app/app-routing.module.ts:14]: We need router guard here also right?

File: src/app/app-routing.module.ts [hunk 9]
```diff
@@ -1,20 +1,21 @@
...
+import { OrganisationGuard, RouteGuard } from './modules/admin-core';
+
 
 
 
 const routes: Routes = [
   {
     path: 'users',
-    canActivate: [RouteGuard],
+    canActivate: [OrganisationGuard],
```
Comment (kiranharidas187) [inline src/app/app-routing.module.ts]: Please give a relevant name

File: src/app/modules/admin-core/guards/organisation.guard.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  async canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot) {
+    this.promiseRowData = await this.usersService.getUserRoles();
+    if (this.promiseRowData['result']) {
+      this.rolesArray = this.promiseRowData['result'].roles;
+    }
+    this.tokendetails = this.keycloakService.sendToken();
+    if (this.rolesArray.includes("PLATFORM_ADMIN") || this.rolesArray.includes("ORG_ADMIN")) {
```
Comment (kiranharidas187) [inline src/app/modules/admin-core/guards/organisation.guard.ts]: OrgGuard should be for organizations page

File: src/app/modules/organisations/create-organisation/create-organisation.component.ts [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+    public dialogRef: MatDialogRef<CreateOrganisationComponent>,
+    private organisationService: OrganisationService,
+    private commonServiceService: CommonServiceService) { }
+
+  ngOnInit() {
+    this.formdata = this.data.fieldsForOrganisation;
+    // this.formdata = [
+    //   {
+    //     "field": "name",
+    //     "value": "srikanth",
```
Comment (kiranharidas187) [inline src/app/modules/organisations/create-organisation/create-organisation.component.ts]: remove commented code

File: src/app/app-routing.module.ts (hunk 8 above)
Comment (srikanthgudipudi) [reply src/app/app-routing.module.ts:14]: No need this guard sufficient

File: src/app/modules/admin-core/guards/organisation.guard.ts (hunk 10 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-core/guards/organisation.guard.ts]: changed to users guard

File: src/app/modules/organisations/create-organisation/create-organisation.component.ts (hunk 11 above)
Comment (srikanthgudipudi) [reply src/app/modules/organisations/create-organisation/create-organisation.component.ts]: Removed

File: src/app/app-routing.module.ts (hunk 9 above)
Comment (srikanthgudipudi) [reply src/app/app-routing.module.ts]: changed

**Pull request merged**